### ISSUE TITLE
Multiple message printing has no separation between messages

### ISSUE THREAD

**Issue by solunetspain** [Zimbra-Community/printpatch-zimlet#3]
Hi, it wors all fine!!! Lots of Thanks for the improvement,  but there's a little question... when you select two or more messages it prints them in the same document, and sometimes impresion is in the same paper sheet. There's no separation between messages. Sometimes you print two or three messages and there's no way to classify them, for example, in different folders in the shelving...
Best regards

**Comment by ghost** [comment]
Something like this:

```
<div style="page-break-after:always">&nbsp;</div>
```

comes to mind... I will look into it later this week.

@jorgedlcruz  fyi

**Comment by ghost** [comment]
Hello,

I have updated the patch to print each mail on a new page, please test once more, please test:

as root:

```
 cp /opt/zimbra/jetty/webapps/zimbra/h/printmessage /opt/zimbra/jetty/webapps/zimbra/h/printmessage-orig
 wget https://raw.githubusercontent.com/Zimbra-Community/printpatch-zimlet/master/printmessage -O /opt/zimbra/jetty/webapps/zimbra/h/printmessage
```

And then as zimbra 

```
 zmmailboxdctl restart
```

**Comment by solunetspain** [comment]
Hi all,
In my case the path was not the same but no biggie

[root @ m2] # locate printmessage
/opt/zimbra/jetty-distribution-9.1.5.v20140505/webapps/zimbra/h/printmessage

The application process is trivial, just rename the current file and replace it with the downloaded (need to change owner and group zimbra in your process as root gives root.root from the file instead of zimbra.zimbra) and restart. All perfectly understandable.

Tested in Chrome, Firefox and Explorer. Sorting by conversation and by date.

It prints  messages in the order that you're selecting. We will test a few days and if a problem occurs inform you but seems to go perfectly.

Zimbra would make sense that it would include in its next update native or the next because it covers important gaps in the print function.

Simply superb job.

## Thank you.

[cid:image001.gif@01D1AB67.BD96E8A0]

Javier García Alarcón
SOLUNET - Soluciones Internet S.L.
Fernando Moliné 40
50171 La Puebla de Alfindén, Zaragoza

## Tf.: 902 10 90 07  Fax: 976 45 55 69

Si Vd. va a imprimir este mensaje, piense primero si es absolutamente necesario. Economizar papel tinta y energía es proteger el medio ambiente
If you're going to print this message, please, think about it. Be careful with the environment.

Este mensaje y sus archivos adjuntos van dirigidos exclusivamente a su destinatario, pudiendo contener información confidencial sometida a secreto profesional. No está permitida su reproducción o distribución sin la autorización expresa de SOLUCIONES INTERNET S.L. Si Vd. no es el destinatario final, por favor elimínelo e infórmenos por esta vía.
                De acuerdo con lo establecido por la Ley Orgánica 15/1999 de 13 de Diciembre, de Protección de Datos de Carácter Personal (LOPD), le informamos que su datos están incorporados en fichero del que es titular SOLUCIONES INTERNET S.L., con la finalidad de realizar la gestión administrativa, contable y fiscal, así como enviarle comunicaciones comerciales sobre nuestros productos y/o servicios.
                Asimismo, le informamos de la posibilidad de ejercer los derechos de acceso, rectificación, cancelación y oposición de sus datos en el domicilio fiscal de SOLUCIONES INTERNET, S.L., sito en c/ Fernando Moliné nº 40 local, 50171 La Puebla de Alfindén (Zaragoza).
                Si Vd. no desea recibir nuestra información, póngase en contacto con nosotros enviado un correo electrónico a la siguiente dirección: facturacion@solunet.esmailto:facturacion@solunet.es

De: Barry de Graaff [mailto:notifications@github.com]
Enviado el: miércoles, 11 de mayo de 2016 8:01
Para: Zimbra-Community/printpatch-zimlet
CC: Javier García Alarcón; Author
Asunto: Re: [Zimbra-Community/printpatch-zimlet] Multiple message printing has no separation between messages (#3)

Hello,

I have updated the patch to print each mail on a new page, please test once more, please test:

as root:

 cp /opt/zimbra/jetty/webapps/zimbra/h/printmessage /opt/zimbra/jetty/webapps/zimbra/h/printmessage-orig

  wget https://raw.githubusercontent.com/Zimbra-Community/printpatch-zimlet/master/printmessage -O /opt/zimbra/jetty/webapps/zimbra/h/printmessage

And then as zimbra

 zmmailboxdctl restart

**Comment by ghost** [comment]
/opt/zimbra/jetty/
is a symlink to...
/opt/zimbra/jetty-distribution-9.1.5.v20140505/

So its the same thing.

Did you file a bug in bugzilla upstream on this?

**Comment by solunetspain** [comment]
Hi Barry

Yes, it is indeed a symbolic link. I started looking for the file and I did not realize it. You're right.

I think it  solves these two bugs in Bugzilla.

https://bugzilla.zimbra.com/show_bug.cgi?id=95856 (there are some posts from me and a couple of attachments)

https://bugzilla.zimbra.com/show_bug.cgi?id=99597

It would be very good idea that  Zimbra incorporate into its core as soon as posible

## Best regards

[cid:image001.gif@01D1AB82.21853050]

Javier García Alarcón
SOLUNET - Soluciones Internet S.L.
Fernando Moliné 40
50171 La Puebla de Alfindén, Zaragoza

## Tf.: 902 10 90 07  Fax: 976 45 55 69

Si Vd. va a imprimir este mensaje, piense primero si es absolutamente necesario. Economizar papel tinta y energía es proteger el medio ambiente
If you're going to print this message, please, think about it. Be careful with the environment.

Este mensaje y sus archivos adjuntos van dirigidos exclusivamente a su destinatario, pudiendo contener información confidencial sometida a secreto profesional. No está permitida su reproducción o distribución sin la autorización expresa de SOLUCIONES INTERNET S.L. Si Vd. no es el destinatario final, por favor elimínelo e infórmenos por esta vía.
                De acuerdo con lo establecido por la Ley Orgánica 15/1999 de 13 de Diciembre, de Protección de Datos de Carácter Personal (LOPD), le informamos que su datos están incorporados en fichero del que es titular SOLUCIONES INTERNET S.L., con la finalidad de realizar la gestión administrativa, contable y fiscal, así como enviarle comunicaciones comerciales sobre nuestros productos y/o servicios.
                Asimismo, le informamos de la posibilidad de ejercer los derechos de acceso, rectificación, cancelación y oposición de sus datos en el domicilio fiscal de SOLUCIONES INTERNET, S.L., sito en c/ Fernando Moliné nº 40 local, 50171 La Puebla de Alfindén (Zaragoza).
                Si Vd. no desea recibir nuestra información, póngase en contacto con nosotros enviado un correo electrónico a la siguiente dirección: facturacion@solunet.esmailto:facturacion@solunet.es

De: Barry de Graaff [mailto:notifications@github.com]
Enviado el: miércoles, 11 de mayo de 2016 10:24
Para: Zimbra-Community/printpatch-zimlet
CC: Javier García Alarcón; Author
Asunto: Re: [Zimbra-Community/printpatch-zimlet] Multiple message printing has no separation between messages (#3)

Closed #3https://github.com/Zimbra-Community/printpatch-zimlet/issues/3.

**Comment by ghost** [comment]
So there is not a specific bug report for the fact Zimbra prints multiple (selected) messages without page breaks.

Anyway, at this time Zimbra will not accept, nor can look at patches legally, so unless a bug is reported and fixed by internal staff, it is not being accepted. That said, the patch uses javascript in a jsp, there is probably nicer way to do it technically. But for us that does not matter,

**Comment by ghost** [comment]
Added https://bugzilla.zimbra.com/show_bug.cgi?id=104927

**Comment by ghost** [comment]
https://bugzilla.zimbra.com/show_bug.cgi?id=104927
Is now confirmed and will be fixed in 8.8

**Comment by solunetspain** [comment]
Thanks for the info Barry  ☺

---

[cid:image001.gif@01D1B271.F24DF960]

Javier García Alarcón
SOLUNET - Soluciones Internet S.L.
Fernando Moliné 40
50171 La Puebla de Alfindén, Zaragoza

## Tf.: 902 10 90 07  Fax: 976 45 55 69

Si Vd. va a imprimir este mensaje, piense primero si es absolutamente necesario. Economizar papel tinta y energía es proteger el medio ambiente
If you're going to print this message, please, think about it. Be careful with the environment.

Este mensaje y sus archivos adjuntos van dirigidos exclusivamente a su destinatario, pudiendo contener información confidencial sometida a secreto profesional. No está permitida su reproducción o distribución sin la autorización expresa de SOLUCIONES INTERNET S.L. Si Vd. no es el destinatario final, por favor elimínelo e infórmenos por esta vía.
                De acuerdo con lo establecido por la Ley Orgánica 15/1999 de 13 de Diciembre, de Protección de Datos de Carácter Personal (LOPD), le informamos que su datos están incorporados en fichero del que es titular SOLUCIONES INTERNET S.L., con la finalidad de realizar la gestión administrativa, contable y fiscal, así como enviarle comunicaciones comerciales sobre nuestros productos y/o servicios.
                Asimismo, le informamos de la posibilidad de ejercer los derechos de acceso, rectificación, cancelación y oposición de sus datos en el domicilio fiscal de SOLUCIONES INTERNET, S.L., sito en c/ Fernando Moliné nº 40 local, 50171 La Puebla de Alfindén (Zaragoza).
                Si Vd. no desea recibir nuestra información, póngase en contacto con nosotros enviado un correo electrónico a la siguiente dirección: facturacion@solunet.esmailto:facturacion@solunet.es

De: Barry de Graaff [mailto:notifications@github.com]
Enviado el: viernes, 20 de mayo de 2016 6:35
Para: Zimbra-Community/printpatch-zimlet
CC: Javier García Alarcón; Author
Asunto: Re: [Zimbra-Community/printpatch-zimlet] Multiple message printing has no separation between messages (#3)

https://bugzilla.zimbra.com/show_bug.cgi?id=104927
Is now confirmed and will be fixed in 8.8

**Comment by ghost** [comment]
This may also be interesting for you:
https://github.com/Zimbra-Community/ca_uoguelph_ccs_printbutton
